src/dune_rules/install_entry.ml

```diff
@@ -13,11 +13,46 @@ let expand_str_with_check_for_local_path ~expand_str sw =
     str)
 ;;
 
+module Glob_files_with_optional_prefix = struct
+  type t =
+    { glob_files : Dep_conf.Glob_files.t
+    ; prefix : String_with_vars.t option
+    }
+
+  let decode =
+    let open Dune_lang.Decoder in
+    let install_glob_version_check = Dune_lang.Syntax.since Stanza.syntax (3, 6) in
+    let install_glob_with_prefix_version_check =
+      Dune_lang.Syntax.since Stanza.syntax (3, 11)
+    in
+    let decode_args ~recursive =
+      let decode_without_prefix =
+        String_with_vars.decode
+        >>| fun glob ->
```

Comment:
nitpick: this style is never an improvement over:

```
let+ glob = String_with_vars.decode in
...
```

The only time we should use `>>|` or `>>=` is whenever we get to do something point-free. Otherwise, it's just making us read `let` bindings in reverse order.